### PR TITLE
[cla] setup the CLA assistant

### LINKED ISSUE #679: [test-infra] Add Contributor License Agreement assistant (CLA assistant)
**Summary**

Like:
ClickHouse: https://yandex.ru/legal/cla/?lang=en
TiDB: https://cla-assistant.io/pingcap/tidb?pullRequest=16303
OceanBase: https://github.com/oceanbase/oceanbase/pull/5

### PR DESCRIPTION
**Pull request by PsiACE**
I hereby agree to the terms of the CLA available at: https://datafuse.rs/policies/cla/

Signed-off-by: Chojan Shang <psiace@outlook.com>

## Summary

Try to add the  "CLA assistant".

- [x] add basic workflow
- [x] add docs, especially the DCLA.md (#736 )
- [ ]  add token by owner, test & sign

## Changelog

- Other 

## Related Issues

Fixes #679 

## Test Plan

Add all files/tokens, try to sign the fist CLA.

## How to setup or sign

WIP

### CODE REVIEW

File: .github/workflows/cla.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  CLAssistant:
+    runs-on: ubuntu-latest
+    steps:
+      - name: "CLA Assistant"
+        if: (github.event.comment.body == 'recheck' || github.event.comment.body == 'I have read the CLA Document and I hereby sign the CLA') || github.event_name == 'pull_request_target'
+        uses: cla-assistant/github-action@v2.1.3-beta
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          # the below token should have repo scope and must be manually added by you in the repository's secret
+          PERSONAL_ACCESS_TOKEN: ${{ secrets.PERSONAL_ACCESS_TOKEN }}
```
Comment (BohuTANG) [inline .github/workflows/cla.yml:18]: What is this PERSONAL_ACCESS_TOKEN  for?

File: .github/workflows/cla.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        with:
+          # You do not need to create this file manually.
+          path-to-signatures: 'cla/signatures/version-1/cla.json'
+          path-to-document: 'https://github.com/datafuselabs/datafuse/blob/master/cla/agreements/DCLA.md'
```
Comment (BohuTANG) [inline .github/workflows/cla.yml:22]: path to `website/datafuse/docs/cla/agreements/DCLA.md`?

Comment (PsiACE) [reply .github/workflows/cla.yml:22]: > path to `website/datafuse/docs/cla/agreements/DCLA.md`?

Why not put everything related to cla together? We can use soft links to provide a copy for docs.

File: .github/workflows/cla.yml (hunk 1 above)
Comment (PsiACE) [reply .github/workflows/cla.yml:18]: > What is this PERSONAL_ACCESS_TOKEN for?

See https://github.com/cla-assistant/github-action#6-adding-personal-access-token-as-a-secret

File: .github/workflows/cla.yml (hunk 2 above)
Comment (BohuTANG) [reply .github/workflows/cla.yml:22]: ok +1

Comment (BohuTANG) [reply .github/workflows/cla.yml:22]: BTW, Clickhouse approach is very simple:

https://github.com/ClickHouse/ClickHouse/blob/master/.github/PULL_REQUEST_TEMPLATE.md

His robot will check the PR contains:
```
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en
```
By retaining this you are agreeing to their CLA and do not need to sign.

Comment (BohuTANG) [reply .github/workflows/cla.yml:22]: image
But the complex is to have status check, I think this patch is to do it.

Comment (PsiACE) [reply .github/workflows/cla.yml:22]: For us, just add a comment like this

```
I have read the CLA Document and I hereby sign the CLA
```

Comment (PsiACE) [reply .github/workflows/cla.yml:22]: For this PR, the expected behavior is just like here: https://github.com/nix-dot-dev/nix.dev/pull/40

**Comment by BohuTANG** [comment]
@PsiACE 

Hi, first thanks for the contribution!
fusebots has add the PR text including detection and it can now detect CLA:
https://github.com/datafuselabs/fusebots/commit/1e47bb1fb8fbada90d43497a4dcc21b420ad1c2d

https://github.com/datafuselabs/datafuse/commit/9b0bfe7baba90f7e016bcab573cff62f01fbae44#diff-18813c86948efc57e661623d7ba48ff94325c9b5421ec9177f724922dd553a35

Demo is here:
https://github.com/datafuselabs/datafuse/pull/691

**Comment by PsiACE** [comment]
> [datafuselabs/fusebots@1e47bb1](https://github.com/datafuselabs/fusebots/commit/1e47bb1fb8fbada90d43497a4dcc21b420ad1c2d)

awesome , but I’m not sure if it can take effect on workflow. I update `github.event.comment.body`, I hope it can work correctly. 

If it fails, we can try another solution with regex: https://github.com/Andrew-Test-Org/Public-Test-Repo/blob/89837fb1176de780c5a574cc195392b398ff01c3/.github/workflows/CLA.yml

> Add CLA in datafuse pull request:
> [9b0bfe7#diff-18813c86948efc57e661623d7ba48ff94325c9b5421ec9177f724922dd553a35](https://github.com/datafuselabs/datafuse/commit/9b0bfe7baba90f7e016bcab573cff62f01fbae44#diff-18813c86948efc57e661623d7ba48ff94325c9b5421ec9177f724922dd553a35)

Thank you very much, I adjusted the relevant path.

> Demo is here:
> #691

This PR also needs to be tested, good luck.

**Comment by BohuTANG** [comment]
hmm, actually we do not need any CLA workflow anymore.
@datafuse-bot will watch the PR update and check the PR's body, then start a Description check, if there is the CLA text in the body, the check will show the  error in the checks list.

**Comment by PsiACE** [comment]
> hmm, actually we do not need any CLA workflow anymore.
> @datafuse-bot will watch the PR update and check the PR's body, then start a Description check, if there is no CLA text in the body, the check will show the error in the checks list.

but @datafuse-bot  does not record the signatures.

**Comment by BohuTANG** [comment]
> > hmm, actually we do not need any CLA workflow anymore.
> > @datafuse-bot will watch the PR update and check the PR's body, then start a Description check, if there is no CLA text in the body, the check will show the error in the checks list.
> 
> but @datafuse-bot does not record the signatures.
Yes, this is not an signed CLA, it likes ClickHouse 'note' CLA:
If the contributor PR keeping the CLA notes, by default they are agreeing to it.

**Comment by PsiACE** [comment]
> Yes, this is not an signed CLA, it likes ClickHouse 'note' CLA:
> If the contributor PR keeping the CLA notes, by default they are agreeing to it.

Okay, I will close this PR. It can be reopened at any time when needed.

**Pull request closed without merging**